app/models/user_mode/auth0_auth_strategy.rb

```diff
@@ -0,0 +1,22 @@
+class UserMode::Auth0AuthStrategy < UserMode::MultiUser
+  def set_auth_provider(omniauth)
+    omniauth.provider :auth0,
+                      Rails.configuration.auth0_client_id,
+                      Rails.configuration.auth0_client_secret,
```

Comment:
I mean, what about having the following structure:

* SingleUserMode: when you have not authentication strategy
* MultiUserMode: when you have authentication. You need to provide an authentication strategy, though: `Saml`, `Auth0`, etc.